spanner/cloud-client/src/main/java/com/example/spanner/LoadCsvExample.java

```diff
@@ -0,0 +1,250 @@
+    } else {
+      throw new IllegalArgumentException(
+          "Unrecognized or unsupported column data type: " + columnType);
+    }
+  }
+
+  public static void parseTableTypes(String tableId) throws SQLException {
+    ResultSet spannerType = connection.createStatement()
+        .executeQuery("SELECT spanner_type FROM information_schema.columns "
+            + "WHERE table_name = \"" + tableId + "\"");
+    while (spannerType.next()) {
+      spannerSchema.add(parseSpannerDataType(spannerType.getString("spanner_type")));
+    }
+  }
+
+  public static void parseTableHeaders(String tableId) throws SQLException {
+    ResultSet tableHeader = connection.createStatement()
+        .executeQuery("SELECT column_name FROM information_schema.columns "
+            + "WHERE table_name = \"" + tableId + "\"");
+    while (tableHeader.next()) {
+      headers.add(tableHeader.getString("column_name"));
+    }
+  }
+
+  public static boolean isValidHeaders(CSVParser parser) {
+    List<String> allHeaders = parser.getHeaderNames();
+    for (int i = 0; i < allHeaders.size(); i++) {
+      if (!allHeaders.get(i).equals(headers.get(i))) {
+        System.err.println(
+            "Header " + allHeaders.get(i) + " does not match database table header " + headers
+                .get(i));
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public static CSVFormat setFormat(CommandLine cmd) {
+    CSVFormat parseFormat;
+    if (cmd.hasOption("f")) {
+      switch (cmd.getOptionValue("f").toUpperCase()) {
+        case (EXCEL):
+          parseFormat = CSVFormat.EXCEL;
+          break;
+        case (POSTGRESQL_CSV):
+          parseFormat = CSVFormat.POSTGRESQL_CSV;
+          break;
+        case(POSTGRESQL_TEXT):
+          parseFormat = CSVFormat.POSTGRESQL_TEXT;
+          break;
+        default:
+          parseFormat = CSVFormat.DEFAULT;
+      }
+    } else {
+      parseFormat = CSVFormat.DEFAULT;
+    }
+
+    if (cmd.hasOption("n")) {
+      parseFormat = parseFormat.withNullString(cmd.getOptionValue("n"));
+    }
+
+    if (cmd.hasOption("d")) {
+      if (cmd.getOptionValue("d").length() != 1) {
+        throw new IllegalArgumentException("Invalid delimiter character entered.");
+      }
+      parseFormat = parseFormat.withDelimiter(cmd.getOptionValue("d").charAt(0));
+    }
+
+    if (cmd.hasOption("e")) {
+      if (cmd.getOptionValue("e").length() != 1) {
+        throw new IllegalArgumentException("Invalid escape character entered.");
+      }
+      parseFormat = parseFormat.withEscape(cmd.getOptionValue("e").charAt(0));
+    }
+
+    if (cmd.hasOption("h") &&  cmd.getOptionValue("h").equalsIgnoreCase("True")) {
+      parseFormat = parseFormat.withFirstRecordAsHeader();
+    }
+    return parseFormat;
+  }
+
+  public static void loadData(Iterable<CSVRecord> records, String tableId)
+      throws SQLException {
+    System.out.println("Writing data into table...");
+    List<Mutation> mutations = new ArrayList<>();
```

Comment:
Is this Mutation used in the JDBC driver too?

Reply:
Not entirely sure if I completely understand what you're asking, but this is the mutation that is passed into the `CloudSpannerJdbcConnection` write method [here](https://github.com/kaylaleung/java-docs-samples/pull/1/files/0b393b38463ac866aed7bf64af4c0593aae6cd45#diff-9c10a7320fc201516b9898e887d21f8d75ea1ccbbb9f1590fcdda512d338ae1dR192) so I think the answer is yes?

Reply:
Makes sense, thanks